src/test/java/dev/vasyl/proj/controller/BookControllerTests.java

```diff
@@ -0,0 +1,249 @@
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class BookControllerTests {
+    protected static MockMvc mockMvc;
+    private static final String CONTROLLER_ENDPOINT = "/books";
+    private static final String DB_PATH_ADD_BOOKS_WITH_CATEGORIES = "database/book/add-books.sql";
+    private static final String CLEAR_BOOKS_AND_CATEGORIES = "database/book/clear-books-and-categories.sql";
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        executeSqlScript(dataSource, CLEAR_BOOKS_AND_CATEGORIES);
+    }
+
+    @BeforeEach
+    void beforeEach(@Autowired DataSource dataSource) {
+        executeSqlScript(dataSource, CLEAR_BOOKS_AND_CATEGORIES);
+        executeSqlScript(dataSource, DB_PATH_ADD_BOOKS_WITH_CATEGORIES);
+    }
+
+    @SneakyThrows
+    static void executeSqlScript(DataSource dataSource, String dbPath) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(dbPath));
+        }
+    }
+
+    @Test
+    @DisplayName("Verify that book is created successfully when requestDto correct")
+    void createBook_correctRequestDto_returnsCreatedBook() throws Exception {
+        CreateBookRequestDto requestDto = getNewCreateBookRequestDto();
+        BookDto expectedBook = getBookDtoByRequestDto(4L, requestDto);
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+        MvcResult result = mockMvc.perform(post(CONTROLLER_ENDPOINT)
+                        .header("Authorization", "Bearer " + getAdminToken())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(jsonRequest))
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actualBook = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto.class);
+        assertNotNull(actualBook.getId());
+        EqualsBuilder.reflectionEquals(expectedBook, actualBook, "id");
```

Comment:
Let's add spaces to make code more readable. See everywhere
```suggestion
        CreateBookRequestDto requestDto = getNewCreateBookRequestDto();
        BookDto expectedBook = getBookDtoByRequestDto(4L, requestDto);
        String jsonRequest = objectMapper.writeValueAsString(requestDto);
        
        MvcResult result = mockMvc.perform(post(CONTROLLER_ENDPOINT)
                        .header("Authorization", "Bearer " + getAdminToken())
                        .contentType(MediaType.APPLICATION_JSON)
                        .content(jsonRequest))
                .andExpect(status().isOk())
                .andReturn();
                
        BookDto actualBook = objectMapper.readValue(
                result.getResponse().getContentAsString(), BookDto.class);
        assertNotNull(actualBook.getId());
        EqualsBuilder.reflectionEquals(expectedBook, actualBook, "id");
```